internal/pkg/deploy/cloudformation/stack/static_site.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/upload/customresource"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+)
+
+// StaticSite represents the configuration needed to create a CloudFormation stack from a static site service manifest.
+type StaticSite struct {
+	*wkld
+	manifest *manifest.StaticSite
+	appInfo  deploy.AppInformation
+
+	parser   staticSiteReadParser
+	localCRs []uploadable // Custom resources that have not been uploaded yet.
+}
+
+// StaticSiteConfig contains fields to configure StaticSite.
+type StaticSiteConfig struct {
+	App                *config.Application
+	EnvManifest        *manifest.Environment
+	Manifest           *manifest.StaticSite
+	RawManifest        []byte // Content of the manifest file without any transformations.
+	RuntimeConfig      RuntimeConfig
+	RootUserARN        string
+	ArtifactBucketName string
+	Addons             NestedStackConfigurer
+}
+
+// NewStaticSite creates a new CFN stack from a manifest file, given the options.
+func NewStaticSite(conf StaticSiteConfig) (*StaticSite, error) {
```

Comment:
nit: to save some mem space
```suggestion
func NewStaticSite(conf *StaticSiteConfig) (*StaticSite, error) {
```